server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -217,4 +246,33 @@ public void deleteSecretByNameAndVersion(String name, String version) {
       }
     });
   }
+
+  public static class SecretDAOFactory implements DAOFactory<SecretDAO> {
+    private final DSLContext jooq;
+    private final DSLContext readonlyJooq;
+    private final SecretContentDAOFactory secretContentDAOFactory;
+    private final SecretSeriesDAOFactory secretSeriesDAOFactory;
+
+    @Inject public SecretDAOFactory(DSLContext jooq, @Readonly DSLContext readonlyJooq,
+        SecretContentDAOFactory secretContentDAOFactory,
+        SecretSeriesDAOFactory secretSeriesDAOFactory) {
+      this.jooq = jooq;
+      this.readonlyJooq = readonlyJooq;
+      this.secretContentDAOFactory = secretContentDAOFactory;
+      this.secretSeriesDAOFactory = secretSeriesDAOFactory;
+    }
+
+    @Override public SecretDAO readwrite() {
+      return new SecretDAO(jooq, secretContentDAOFactory, secretSeriesDAOFactory);
+    }
+
+    @Override public SecretDAO readonly() {
+      return new SecretDAO(readonlyJooq, secretContentDAOFactory, secretSeriesDAOFactory);
+    }
+
+    @Override public SecretDAO using(Configuration configuration) {
```

Comment:
When you're calling a DAO from another DAO in a transaction, the caller would have to use `SomethingDAOFactory.using(DSL.using(configuration))` since you get a Configuration from jOOQ in a transaction.